xcm/xcm-executor/src/lib.rs

```diff
@@ -139,7 +226,18 @@ impl<Config: config::Config> ExecuteXcm<Config::Call> for XcmExecutor<Config> {
 			}
 		}
 
-		vm.post_execute(xcm_weight)
+		vm.post_process(xcm_weight)
+	}
+
+	fn charge_fees(origin: impl Into<MultiLocation>, fees: MultiAssets) -> XcmResult {
+		let origin = origin.into();
+		if !Config::FeeManager::is_waived(Some(&origin), FeeReason::ChargeFees) {
+			for asset in fees.inner() {
+				Config::AssetTransactor::withdraw_asset(&asset, &origin, None)?;
+			}
+			Config::FeeManager::handle_fee(fees);
```

Comment:
Am I right that the xcm router (`type XcmRouter: SendXcm;`) is unable to influence on what happens with its fees? If .e.g it'll be configured to `type XcmRouter = (ChildParachainRouter, OtherRelayChainBridgeHubRouter);`, then my guess it that the fees paid for messages routed using `ChildParachainRouter` may be burnt or distributed across validators/collators/treasury. But  for `OtherRelayChainBridgeHubRouter`, we'll need these fees to compensate operational costs of our relayers - e.g. relayer pays for the delivery transaction at the bridged chain.

E.g. in all current (possibly intermediate) configurations, `FeeManager` is set to `()`, which means that all fees will be burnt? Shall we then just mint tokens to reward relayers?

Reply:
Basically, yes.

The most recent thinking on relayer fees is that they should be paid by the treasury in any case.